test/v2_integration.py

```diff
@@ -977,5 +977,32 @@ def test_long_san_no_cn():
         if e.detail != 'Error finalizing order :: issuing precertificate: CSR doesn\'t contain a SAN short enough to fit in CN':
             raise Exception('Problem detail did not match expected')
 
+def test_delete_unused_challenges():
+    if not CONFIG_NEXT:
+      return
+    order = chisel2.auth_and_issue([random_domain()], chall_type="dns-01")
+    for a in order.authorizations:
+        if len(a.body.challenges) != 1:
+            raise Exception("too many challenges (%d) left after validation", len(a.body.challenges))
+        if not isinstance(a.body.challenges[0].chall, challenges.DNS01):
+            raise Exception("wrong challenge type left after validation")
+
+    # intentionally fail a challenge
+    client = chisel2.make_client()
+    csr_pem = chisel2.make_csr([random_domain()])
+    order = client.new_order(csr_pem)
+    c = get_chall(a, chall_type)
+    client.answer_challenge(c, c.response(client.net.key))
+    try:
+        client.poll_and_finalize(order)
+    except errors.ValidationError as e:
+        pass
+    for a in order.authorizations:
```

Comment:
Presumably you only want to do this for the authorization that you failed rather than all the authorizations in the order.

Reply:
(not sure myself but: does `poll_and_finalize` update the `order` you pass in? or do you need to re-fetch it yourself/fetch the authorization)

Reply:
I thought it did but I looked a second time and it does not. It does return an updated error, but only in non-failure cases. Rewrote to make it work.